database/migrations/community/2024_03_23_000000_update_setclaim_table.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        // [1] Rename GameID to game_id, establish foreign key relationship to GameData table.
+        Schema::table('SetClaim', function (Blueprint $table) {
+            $table->unsignedBigInteger('GameID')->change();
+        });
+        Schema::table('SetClaim', function (Blueprint $table) {
+            $table->renameColumn('GameID', 'game_id');
+        });
+        Schema::table('SetClaim', function (Blueprint $table) {
+            $table->foreign('game_id')->references('ID')->on('GameData')->onDelete('cascade');
+        });
+
+        // [2] Add user_id, establish foreign key relationship to UserAccounts table.
+        Schema::table('SetClaim', function (Blueprint $table) {
+            $table->unsignedBigInteger('user_id')->nullable()->after('user');
+        });
+        Schema::table('SetClaim', function (Blueprint $table) {
+            $table->foreign('user_id')->references('ID')->on('UserAccounts')->onDelete('cascade');
+        });
+    }
+
+    public function down(): void
+    {
+        // [2] Add user_id, establish foreign key relationship to UserAccounts table.
+        Schema::table('SetClaim', function (Blueprint $table) {
+            $table->dropForeign(['user_id']);
+            $table->dropColumn('user_id');
+        });
+
+        // [1] Rename GameID to game_id, establish foreign key relationship to GameData table.
+        Schema::table('SetClaim', function (Blueprint $table) {
+            $table->dropForeign(['game_id']);
+            $table->renameColumn('game_id', 'GameID');
+            $table->unsignedInteger('GameID')->change();
+        });
```

Comment:
My rollback failed. Not sure why - this _looks_correct.
```
2024_03_23_000000_update_setclaim_table ............................................... 40ms FAIL

   Doctrine\DBAL\Schema\Exception\ColumnDoesNotExist 

  There is no column with name "gameid" on table "SetClaim".
```

Reply:
I had the same issue but assumed it was something up with my local machine, as the error is a bit bizarre.

I was able to fix this on my local after deconstructing every single operation into separate `Schema::table()` tasks and forcing Doctrine to identify `game_id` as an `unsignedBigInteger` before the rename runs. Now that rollback seems to work correctly:

```php
Schema::table('SetClaim', function (Blueprint $table) {
    $table->dropForeign(['game_id']);
});
Schema::table('SetClaim', function (Blueprint $table) {
    $table->unsignedBigInteger('game_id')->change();
});
Schema::table('SetClaim', function (Blueprint $table) {
    $table->renameColumn('game_id', 'GameID');
});
Schema::table('SetClaim', function (Blueprint $table) {
    $table->unsignedInteger('GameID')->change();
});
```